.markdownlint.yaml

```diff
@@ -0,0 +1,19 @@
+# Project settings for markdownlint behavior.
```

Comment:
Seems like this is enforcing h1 as the first header in all markdown files. That's going to cause a lot of problems as we'll have fragments that will likely start with a variety of headings (or none at all). Can we change this in settings?

Reply:
If/when we add fragments, that is very easy to accomodate for using `-i  path/to/fragments` in the linter command.

It really is best to use the recommended header enforcement for other files. Part of the benefit of this keeps things objective and contributors know what to expect.

Reply:
Going to drop from this PR for now until we can have more discussions as a team. There are a lot of other benefits to adding this mdlint check, so will follow up with any header changes once we have some consensus on things.

Reply:
@stmcginnis This makes sense if you're talking about standard markdown files in a repo (contributing, readme, etc.). 

This doesn't make sense in a static site generator - every markdown file is, in effect, a fragment. So the outer layout will have the site name as h1 and then if we enforce `#` as the first line in the markdown file then we'll have two `h1` tags which breaks the heading hierarchy out of the box. If we need to do this on occasion (for imported file, as example), ok, but having two H1 tags on every page seems like poor page construction. 

Additionally, we'll likely have a variety of markdown files that don't result in a page (literal fragments). Changing the linting rules for multiple directories breaks the spirit of linting in my opinion. 

If you take a look at [Karpenter](https://karpenter.sh/), which is also a Hugo site, they don't enforce this and [have](https://github.com/aws/karpenter/blob/main/website/content/en/v0.16.1/faq.md) [various](https://github.com/aws/karpenter/blob/main/website/content/en/v0.16.1/AWS/launch-templates.md) [ways](https://github.com/aws/karpenter/blob/main/website/content/en/v0.16.1/tasks/set-resource-limits.md) of handling heading hierarchy.